cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationSetReader.java

```diff
@@ -1,21 +1,170 @@
+      }
+    }
+
+    this.contextRoot = configRoot;
+
+    readConfiguration();
+  }
+
+  /**
+   * search for configuration files in the subfolder for adapted templates
+   *
+   * @param configRoot root directory of the configuration template-sets/adapted
+   */
+  private List<Path> loadContextFilesAdapted(Path configRoot) {
+
+    List<Path> contextPaths = new ArrayList<>();
+
+    List<Path> templateDirectories = new ArrayList<>();
+
+    try (Stream<Path> files = Files.list(configRoot)) {
+      files.forEach(path -> {
+        if (Files.isDirectory(path)) {
+          templateDirectories.add(path);
+        }
+      });
+    } catch (IOException e) {
+      throw new InvalidConfigurationException(configRoot, "Could not read configuration root directory.", e);
+    }
+
+    for (Path templateDirectory : templateDirectories) {
+      Path contextPath = templateDirectory.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER)
+          .resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+      if (Files.exists(contextPath)) {
+        contextPaths.add(contextPath);
+        this.configRoots.put(contextPath, templateDirectory);
+      }
+    }
+
+    return contextPaths;
   }
 
+  /**
+   * search for configuration files in the subfolder for downloaded template jars
+   *
+   * @param configRoot root directory of the configuration template-sets/downloaded
+   */
+  private List<Path> loadContextFilesDownloaded(Path configRoot) {
+
+    List<Path> contextPaths = new ArrayList<>();
+
+    List<Path> templateJars = new ArrayList<>();
+
+    try (Stream<Path> files = Files.list(configRoot)) {
+      files.forEach(path -> {
+        if (path.toUri().toString().endsWith(".jar")) {
+          templateJars.add(path);
+        }
+      });
+    } catch (IOException e) {
+      throw new InvalidConfigurationException(configRoot, "Could not read configuration root directory.", e);
+    }
+
+    for (Path jarPath : templateJars) {
+      Path configurationPath = FileSystemUtil.createFileSystemDependentPath(jarPath.toUri());
+      Path contextPath = configurationPath.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER)
+          .resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+
+      if (Files.exists(contextPath)) {
+        contextPaths.add(contextPath);
+        this.configRoots.put(contextPath, configurationPath);
+      }
+    }
+
+    return contextPaths;
+  }
+
+  @Override
+  public Map<String, Trigger> loadTriggers() {
+
+    Map<String, Trigger> triggers = Maps.newHashMap();
+    for (Path contextFile : this.contextConfigurations.keySet()) {
+      Path configRoot = this.configRoots.get(contextFile);
+      ContextConfiguration contextConfiguration = this.contextConfigurations.get(contextFile);
+      boolean isJarConfig = (configRoot.getParent() == null);
+      for (com.devonfw.cobigen.impl.config.entity.io.Trigger t : contextConfiguration.getTrigger()) {
```

Comment:
with the new configuration just one trigger is allowed. maybe we should should use just the first to be compliant with the get Trigger method but make sure we don't have any context.xml files which use more than one trigger.

Reply:
Changed it so that only the first entry is used. Do you mean that also an error should be thrown if there is more than one trigger in a context config file?